src/test/java/teammates/ui/webapi/UpdateFeedbackSessionActionTest.java

```diff
@@ -94,6 +112,268 @@ protected void testExecute() {
+        long endTimePlus1Day = endTime.plus(Duration.ofDays(1)).toEpochMilli();
+        long endTimePlus2Days = endTime.plus(Duration.ofDays(2)).toEpochMilli();
+
+        loginAsInstructor(instructor1ofCourse1.getGoogleId());
+
+        String[] param = new String[] {
+                Const.ParamsNames.COURSE_ID, session.getCourseId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, session.getFeedbackSessionName(),
+                Const.ParamsNames.MUST_NOTIFY_ABOUT_DEADLINES, String.valueOf(false),
+        };
+
+        ______TS("create new deadline extension for student");
+
+        assertNull(expectedStudentDeadlines.get("student1InCourse1@gmail.tmt"));
+
+        FeedbackSessionUpdateRequest updateRequest = getTypicalFeedbackSessionUpdateRequest();
+        Map<String, Long> newStudentDeadlines = convertDeadlinesToLong(updateRequest.getStudentDeadlines());
+        newStudentDeadlines.put("student1InCourse1@gmail.tmt", endTimePlus1Day);
+        updateRequest.setStudentDeadlines(newStudentDeadlines);
+
+        UpdateFeedbackSessionAction a = getAction(updateRequest, param);
+        JsonResult r = getJsonResult(a);
+        FeedbackSessionData response = (FeedbackSessionData) r.getOutput();
+
+        expectedStudentDeadlines.put("student1InCourse1@gmail.tmt", endTimePlus1Day);
+        assertEquals(expectedStudentDeadlines, response.getStudentDeadlines());
+
+        ______TS("update deadline extension for student");
+
+        assertNotEquals(endTimePlus2Days, expectedStudentDeadlines.get("student1InCourse1@gmail.tmt"));
+
+        updateRequest = getTypicalFeedbackSessionUpdateRequest();
+        newStudentDeadlines = convertDeadlinesToLong(updateRequest.getStudentDeadlines());
+        newStudentDeadlines.put("student1InCourse1@gmail.tmt", endTimePlus2Days);
+        updateRequest.setStudentDeadlines(newStudentDeadlines);
+
+        a = getAction(updateRequest, param);
+        r = getJsonResult(a);
+        response = (FeedbackSessionData) r.getOutput();
+
+        expectedStudentDeadlines.put("student1InCourse1@gmail.tmt", endTimePlus2Days);
+        assertEquals(expectedStudentDeadlines, response.getStudentDeadlines());
+
+        ______TS("delete deadline extension for student");
+
+        assertNotNull(expectedStudentDeadlines.get("student1InCourse1@gmail.tmt"));
+
+        // The typical update request does not contain the course 1 student 1's email.
+        updateRequest = getTypicalFeedbackSessionUpdateRequest();
+
+        a = getAction(updateRequest, param);
+        r = getJsonResult(a);
+        response = (FeedbackSessionData) r.getOutput();
+
+        // The deadline for course 1 student 1 was deleted; the map no longer contains a deadline for them.
+        expectedStudentDeadlines.remove("student1InCourse1@gmail.tmt");
+        assertEquals(expectedStudentDeadlines, response.getStudentDeadlines());
+
+        ______TS("C_UD deadline extensions for students simultaneously");
```

Comment:
`simultaneously` might not be an appropriate word to use here as it implies there exists deadline extension in the request where the extension is being created, updated and removed all at once. Perhaps `C_UD deadlines for multiple students` or `typical deadline extension request` will be better?

On the same note, how would the endpoint handle the case where the same deadline extension is updated and removed? Not sure if it will ever happen from the frontend.

Reply:
>`simultaneously` might not be an appropriate word to use here as it implies there exists deadline extension in the request where the extension is being created, updated and removed all at once

Yea, I also thought so, but I wasn't completely sure how else to describe it. Not very sure if either `C_UD deadlines for multiple students` or `typical deadline extension request` completely describes that the three operations are happening within the same request (though they _do_ show the operations are on different extensions/for different users), though I also can't think of any other description.

I suppose I can write `C_UD on different respective deadline extensions within the same request`. Not sure if that is verbose. Though it might be suitable, what do you think?

Reply:
`C_UD on extensions for different users within the same request` might be better.

Reply:
Ok, sure!